app/assets/javascripts/user-actions.js

```diff
@@ -0,0 +1,7 @@
+$(document).ready(function() {
+  $('.add-favorite-button').on('submit', ajaxButtonToggle);
+  $('.remove-favorite-button').on('submit', ajaxButtonToggle);
+
+  $('.add-visit-button').on('submit', ajaxButtonToggle);
+  $('.remove-visit-button').on('submit', ajaxButtonToggle);
+});
```

Comment:
Might be dope to use a naming convention like `class="ajax-button"` and then instead of writing out a listener for all these we could be like
```javascript
$('.toggle-button').forEach(function (button) {
  button.on('submit', ajaxButtonToggle);
});
```